thunder/dynamo/compiler.py

```diff
@@ -12,6 +12,14 @@ def _warn_thunder_compiler():
     )
 
 
+def recompile_graph(gm: torch.fx.GraphModule):
+    # NOTE - `gm` could also be the `_LazyGraphModule`, in which case calling `recompile` is not enough as it marks the `GraphModule`
+    # and actual recompilation happens when either `forward` or `code` (or when user tries to observe the GraphModule).
+    # See for more details - https://github.com/pytorch/pytorch/blob/39935e0fdef02c67ba808175dcc800d0695bfe1b/torch/fx/_lazy_graph_module.py#L65-L89
+    gm.recompile()
+    gm.code
```

Comment:
Other option is 

```python
if isinstance(gm, torch.fx._lazy_graph_module._LazyGraphModule):
     return gm.real_recompile()
return gm.recompile()
```

Reply:
And one more variant to choose from:
```py
from torch.fx._lazy_graph_module import _LazyGraphModule
_LazyGraphModule.force_recompile(gm)
```

Reply:
Accessing code is a clever solution because real_recompile is triggered in _LazyGraphModule 
https://github.com/pytorch/pytorch/blob/39935e0fdef02c67ba808175dcc800d0695bfe1b/torch/fx/_lazy_graph_module.py#L176 but it's better to rely on explicit calls to do recompilation so it's better to choose the code snippet you posted or the one that I posted.